code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/code_generation/BuildScriptCodeGenerator.xtend

```diff
@@ -0,0 +1,99 @@
+ */
+@CodeGenerationModule(name="buildscript")
+class BuildScriptCodeGenerator extends AbstractCodeGenerationModule {
+    var IntermediateOperationModel model
+    var Map<String, String> content
+
+    /*
+     * This method is responsible for handling the instance of the intermediate operation model.
+     * The corresponding functions are called depending on whether the operation node is an instance
+     * of an intermediate container or infrastructure node.
+     */
+    override execute(String[] arg0, String[] arg1) {
+        // Receive the intermediate operation model
+        model = this.intermediateModelResource.contents.get(0) as IntermediateOperationModel
+
+        // Create a Map which contains the generated source code
+        content = new HashMap
+
+        OpendRootBuildScript.instance.loadExistingRootBuildScript(targetFolder)
+
+        // Create a build script for containers
+        model.containers.forEach[container |
+            val nodeName = Util::buildPathFromQualifiedName(
+                container.deployedServices?.get(0).qualifiedName
+            )
+            this.createNodeSpecificBuildScripts(nodeName.toLowerCase, container.name.toLowerCase)
+            this.createRootBuildScript(nodeName.toLowerCase)
+        ]
+
+        // Create build script for infrastructure nodes
+        model.infrastructureNodes.forEach[node |
+            val property = node.defaultValues.findFirst[property |
+                property.technologySpecificProperty.name == "imageType"]
+            val nodeName = node.name.toLowerCase
+
+            if (property.value.toLowerCase != "dockerhub") {
+                this.createNodeSpecificBuildScripts(nodeName, nodeName)
+                this.createRootBuildScript(node.name.toLowerCase)
+            }
+        ]
+
+        content.put(targetFolder + File.separator + "build" + File.separator +  "build.sh",
+            OpendRootBuildScript.instance.toString
+        )
+
+        println("CodeGenerationModule buildscript finished.")
+        return withCharset(content, "UTF-8");
+    }
+
+    /*
+     * Receive the language description for the intermediate operation model package.
+     */
+    @NotNull
+    override getLanguageDescription() {
+        new LanguageDescription(IntermediatePackage.eINSTANCE)
+    }
+
+    /*
+     * Create an operation node specific build script.
+     */
+    private def createNodeSpecificBuildScripts(String nodeName, String containerName) {
+        content.put(targetFolder
+            + File.separator
+            + nodeName
+            + File.separator
+            + "build.sh",
+            BuildScriptTemplate::getNodeSpecificBuildScript(containerName)
+        )
+    }
+
+    /*
+     * Create a root build script that uses the node specific build scripts to build all operation
+     * nodes present in the intermediate operation model.
+     */
+    private def createRootBuildScript(String nodeName) {
```

Comment:
If I got it right, this method is invoked only once in the class and consists of an easy-to-grasp statement. I suggest to remove the method and replace its call with the statement.

Reply:
The method is called twice in the class. Once for all containers of the model and once for all infrastructure nodes.